ocean_lib/models/test/test_data_nft_factory.py

```diff
@@ -330,10 +331,18 @@ def test_nonexistent_template_index(data_nft_factory, publisher_wallet):
     )
     assert non_existent_nft_template >= 0, "Non existent NFT template not found."
 
-    with pytest.raises(Exception, match="Template index doesnt exist"):
-        data_nft_factory.create(
+    with pytest.raises(ValueError, match="Template index doesnt exist"):
```

Comment:
why do we still need interrogation? You needed to interrogate the chain and recheck the transaction status *only on the assumption* that pytest.raises did not catch these. Only then we needed to confirm

Reply:
Because it doesn't use a contract view function such as `getTokenTemplate`, it uses `deployERC721Contract` which is external/payable.

Reply:
I don't understand. So pytest.raises works, it sees the error (now, with required confs set to 0). That is enough for us to consider the tx failed. No need to do the sleep and interrogation.

Reply:
and same everywhere... No need to confirm the transaction failed if it raises the error. The problem was it was hanging instead of failing. Now that it properly fails (as indicated by pytest.raises), there is no need to confirm that it failed on-chain.

Reply:
So, I create a wrong tx and in order to extract some values from it, I need to wait to be completely settled, otherwise I get weird errors from brownie such as `TypeError: int() can't convert non-string with explicit base`. 
`pytest.raises` works now because of that interrogation of blockchain `web3.eth.call()` from where I could retrieve the error message.
Between consecutive exception handling blocks, I need to put a sleep, otherwise the test fails with this error `TypeError: int() can't convert non-string with explicit base`.
As far as I saw from their code, they use a thread to make those transactions and maybe this can be a synchronization problem.

Reply:
That error ocurrs at this line `step["pc"] = int(step["pc"], 16)` and basically `step["pc"]` is not fetched properly from the transaction.